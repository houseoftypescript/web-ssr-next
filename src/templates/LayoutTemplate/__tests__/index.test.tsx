import { render } from '@testing-library/react';
import { LayoutTemplate } from '..';

describe('LayoutTemplate', () => {
  it('to match snapshot without children', () => {
    const { container } = render(<LayoutTemplate />);
    expect(container).toMatchSnapshot();
  });

  it('to match snapshot with children', () => {
    const { container } = render(
      <LayoutTemplate>
        <div>Layout</div>
      </LayoutTemplate>
    );
    expect(container).toMatchSnapshot();
  });
});
