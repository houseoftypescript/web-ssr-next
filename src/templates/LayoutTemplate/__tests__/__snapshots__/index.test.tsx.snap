// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutTemplate to match snapshot with children 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col"
  >
    <nav>
      <div
        class="py-8 border-b"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="flex justify-between items-center"
          >
            <div
              class="flex items-center gap-4 uppercase"
            >
              <h1
                class="text-2xl"
              >
                <a
                  href="/apps"
                >
                  Solid
                </a>
              </h1>
              <a
                href="/apps/chart"
              >
                Chart
              </a>
              <a
                href="/apps/table"
              >
                Table
              </a>
              <a
                href="/apps/calendar"
              >
                Calendar
              </a>
            </div>
            <div>
              <div>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                    data-testid="AccountCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="grow"
    >
      <div>
        Layout
      </div>
    </main>
    <footer>
      <div
        class="py-8 border-t"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <p>
            © 2023 Solid
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;

exports[`LayoutTemplate to match snapshot without children 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col"
  >
    <nav>
      <div
        class="py-8 border-b"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="flex justify-between items-center"
          >
            <div
              class="flex items-center gap-4 uppercase"
            >
              <h1
                class="text-2xl"
              >
                <a
                  href="/apps"
                >
                  Solid
                </a>
              </h1>
              <a
                href="/apps/chart"
              >
                Chart
              </a>
              <a
                href="/apps/table"
              >
                Table
              </a>
              <a
                href="/apps/calendar"
              >
                Calendar
              </a>
            </div>
            <div>
              <div>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                    data-testid="AccountCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="grow"
    />
    <footer>
      <div
        class="py-8 border-t"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <p>
            © 2023 Solid
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
